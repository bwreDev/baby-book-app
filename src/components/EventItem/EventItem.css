.event-item {
  text-transform: capitalize;
}
