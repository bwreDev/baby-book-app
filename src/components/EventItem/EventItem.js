import React from 'react';
import moment from 'moment';
import './EventItem.css';

export default function EventItem(props) {
  const cleanDate = moment(props.event.date_added).format(
    'MMM Do YY, h:mm a'
  );

  return (
    <>
      <h3 className='event-item'>{props.event.content}</h3>
      <span>Posted: {props.event.date_added ? cleanDate : ''}</span>
    </>
  );
}

EventItem.defaultProps = {
  event: {},
};
