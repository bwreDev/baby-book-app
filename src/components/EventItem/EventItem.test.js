import React from 'react';
import ReactDOM from 'react-dom';
import EventItem from './EventItem';
import renderer from 'react-test-renderer';

it('should render without error', () => {
  const div = document.createElement('div');
  ReactDOM.render(<EventItem />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders the UI as expected', () => {
  const tree = renderer.create(<EventItem />).toJSON();
  expect(tree).toMatchSnapshot();
});
