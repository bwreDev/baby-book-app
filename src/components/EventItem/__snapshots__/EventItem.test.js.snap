// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
Array [
  <h3
    className="event-item"
  />,
  <span>
    Posted: 
    
  </span>,
]
`;
